go/vt/sqlparser/ast.go

```diff
@@ -1365,6 +1365,7 @@ type ColumnType struct {
 // ColumnTypeOptions are generic field options for a column type
 type ColumnTypeOptions struct {
 	NotNull       bool
+	Null          bool
```

Comment:
do we need both of them?

Reply:
Yes we need both of them to distinguish 3 cases - 
1. When Not Null is specified
2. When Null is specified
3. When nothing is specified

cases 1 and 2 are straightforward. The complexity arises from the fact that we do not know whether the column will be nullable or not if nothing is specified. We can have a function that returns the default behavior of the data type of the column but otherwise we would require both the variables.